discovery_engine_core/web-api/src/bin/ingestion.rs

```diff
@@ -257,17 +275,24 @@ async fn handle_add_data(
         .body(bytes)
         .send()
         .await
-        .map_err(handle_elastic_error)?
+        .map_err(|err| warp::reject::custom(Error::Elastic(err)))?
         .error_for_status()
-        .map_err(handle_elastic_error)?
+        .map_err(|err| warp::reject::custom(Error::Receiving(err)))?
         .json::<HashMap<String, serde_json::Value>>()
         .await
         .map_err(|err| {
             error!("ReceivingOpError {:#?}", err);
-            warp::reject::custom(ReceivingOpError)
+            warp::reject::custom(Error::Elastic(err))
         })?;
 
-    Ok(StatusCode::OK)
+    if !errored_ids.is_empty() {
+        // TODO: change name of error variant to take into account the elastic error indices
```

Comment:
elastic errors are covered by #647, but it doesn't contain a custom error yet, it depends which PR will me merged sooner.